### ISSUE TITLE
Question: How to apply my flash to all controller in policies.js. Need to apply it to all controller and action and without adding this for each action

### ISSUE THREAD

**Issue by echorohit** [balderdashy/sails#3132]
`` module.exports.policies = {
    
    '*': [true, 'flash', 'guestCheckout', 'setGaData', 'fetchHeaderAndFooter' , 'setSeo'],
    
    CheckoutController: {
        '*': ['hasItemInCart', 'flash', 'setSeo', 'setGaData'],
        'loadAddress': ['sessionAuth'],
        'reviewOrders': ['sessionAuth'],
        'success': ['hasItemInCart', 'setSeo', 'setGaData'],
        'loadPaymentOptions': ['sessionAuth'],
        'addShippingAddress': ['validateShippingAddress'],
        'placeCreditDebitOrder': ['validateCreditDebit'],
        'validatePayback': ['validatePayback'],
        'applyPayback': ['validatePayback'],
        'orderFeedback': 'flash'
    },
    CartController: {
        'index': ['guestCheckout', 'setGaData', 'flash', 'setSeo'],
        'addGiftWrap': ['validateGiftWrap'],
        'validatePayback': ['validatePayback'],
        'applyPayback': ['validatePayback']
    },
    AccountController: {
        '*': ['flash', 'sessionAuth', 'fetchHeaderAndFooter', 'setSeo','setGaData', 'fetchPixelData', 'fetchOffer'],
        'voucher': ['sessionAuth', 'setGaData'],
        'updateAddress': ['sessionAuth', 'validateAddress', 'setGaData'],
        'addAddress': ['sessionAuth', 'validateAddress', 'setGaData'],
        'saveNextBeeCustomerType': ['flash', 'setGaData'],
        'saveNextBeePoints': ['flash', 'setGaData'],
        'footerNewsletterSubscription': ['flash', 'setGaData']
    },
    AuthController: {
        '*': ['flash', 'forbidIfLoggedIn', 'setSeo'],
        'logout': ['flash', 'sessionAuth'],
        'postJbSignup': 'validateSignup',
        'fbSignUp': 'flash',
        'fbSignUpCallBack': 'flash',
        'googleSignUp': 'flash',
        'googleSignUpCallBack': 'flash'
    },
    IndexController: {
        'index': ['guestCheckout', 'setGaData', 'fetchHeaderAndFooter', 'flash', 'setSeo', 'fetchOffer']
    },
    PDPController: {
        'index': ['guestCheckout', 'setGaData', 'fetchHeaderAndFooter', 'flash', 'setSeo', 'fetchOffer', 'linkify']
    },
    CatalogController: {
        'index': ['guestCheckout', 'setGaData', 'fetchHeaderAndFooter', 'flash', 'setSeo', 'fetchOffer', 'redirect', 'campaignBanner']
    },
    CmsController: {
        '*': ['fetchHeaderAndFooter', 'fetchPixelData', 'fetchOffer', 'setGaData']
    },
    ExchangeController: {
        '*': ['flash', 'sessionAuth', 'setGaData']
    },
    OrderController: {
        '*': ['sessionAuth', 'flash', 'fetchHeaderAndFooter', 'flash', 'setGaData', 'setSeo'],
        'track': ['flash', 'setGaData', 'setSeo'],
        'trackOrder': ['flash', 'setGaData', 'setSeo'],
        'view': ['fetchHeaderAndFooter', 'flash', 'setGaData', 'setSeo']
    }
} ``

This is  My 'config/policies.js 

my flash.js file is

`` module.exports = function(req, res, next) {
    initMiddlewareApiTime(req, "policy:flash")
    res.locals.flash = function() {
        var result = req.flash();
        if (!Object.keys(result).length) {
            return false;
        } else {
            return joinAndSingletMessages(result);
        }
    };
    logMiddlewareApiTime(req, "policy:flash")
    next();
 };
 
function joinAndSingletMessages(result) {
    if (result.error) {
        result.error = _.uniq(result.error);
        result.error = [result.error.join(' ')];
    } else if (result.success) {
        result.success = _.uniq(result.success);
        result.success = [result.success.join(' ')];
    }
    return result;
 } 
``

**Comment by Josebaseba** [comment]
Add a middleware that does the stuff you want. More info: http://sailsjs.org/documentation/concepts/middleware

**Comment by sailsbot** [comment]
Thanks for posting, @echorohit.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
